presto-raptorx/src/main/java/com/facebook/presto/raptorx/systemtable/ChunkSystemTable.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.raptorx.util.PredicateUtil.getColumnIndex;
+import static com.facebook.presto.raptorx.util.PredicateUtil.getStringValue;
+import static com.facebook.presto.raptorx.util.PredicateUtil.listTables;
+import static com.facebook.presto.spi.SystemTable.Distribution.SINGLE_COORDINATOR;
+import static com.facebook.presto.spi.predicate.TupleDomain.extractFixedValues;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ChunkSystemTable
+        implements SystemTable
+{
+    private final TransactionManager transactionManager;
+    private final ConnectorTableMetadata tableMetadata;
+
+    @Inject
+    public ChunkSystemTable(TransactionManager transactionManager, TypeManager typeManager)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+
+        this.tableMetadata = new ConnectorTableMetadata(
+                new SchemaTableName("system", "chunks"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("table_schema", VARCHAR))
+                        .add(new ColumnMetadata("table_name", VARCHAR))
+                        .add(new ColumnMetadata("chunk_id", BIGINT))
+                        .add(new ColumnMetadata("bucket_number", BIGINT))
+                        .add(new ColumnMetadata("uncompressed_size", BIGINT))
+                        .add(new ColumnMetadata("compressed_size", BIGINT))
+                        .add(new ColumnMetadata("row_count", BIGINT))
+                        .add(new ColumnMetadata("xxhash64", createVarcharType(16)))
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        Transaction transaction = transactionManager.get(transactionHandle);
+
+        Map<Integer, NullableValue> values = extractFixedValues(constraint).orElse(ImmutableMap.of());
+        Optional<String> schemaName = getStringValue(values.get(getColumnIndex(tableMetadata, "table_schema")));
+        Optional<String> tableName = getStringValue(values.get(getColumnIndex(tableMetadata, "table_name")));
+
+        PageListBuilder builder = PageListBuilder.forTable(tableMetadata);
+
+        Iterator<Page> iterator = listTables(transaction, schemaName, tableName).stream()
+                .map(table -> {
```

Comment:
It might make sense moving this part into `IteratorPageSource` and do the computation on calling `getNextPage`...